### PR TITLE
Add a task which removes all Submissions and related data

### LINKED ISSUE #585: delete test trial submissions
I thought you deleted current test submissions. 
However, I can still see content related to trials and populations ( but not the scores themselves) when looking up "ASSYST". 
I recently submitted "ASSYST accessions" which, of course also come up. Could you remove any entry that has "ASSYST" in their fields which are searchable via the website's search field?

### PR DESCRIPTION
**Pull request by nowakowski**
Fixes #585

### CODE REVIEW

**Comment by nowakowski** [comment]
@Nuanda After we merge this branch, I'll need it deployed on prod so I can back up the DB and run the script - please let me know when it becomes available and I'll take care of the rest.

**Comment by Nuanda** [comment]
@kammerer Please review this one.

**Comment by nowakowski** [comment]
Seems the Travis ES bug is rearing its ugly head again.

File: lib/tasks/clean_up_submissions.rake [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+        records.each do |r|
+          r.delete
+        end
+      else
+        puts "#{m.name} has no user_id attribute. Skipping."
+      end
+    end
+
+    puts "Rebuilding ES indices..."
+    %x( curl -XPOST 'http://localhost:9200/_refresh' )
```
Comment (kammerer) [inline lib/tasks/clean_up_submissions.rake]: ES url should probably be taken from the configuration.

File: lib/tasks/clean_up_submissions.rake [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+    sus = Submission::Upload.all
+    puts "Found #{sus.length} submission uploads."
+
+    sus.each do |su|
+      su.destroy
+    end
+
+    models = ActiveRecord::Base.descendants
+
+    models.each do |m|
```
Comment (kammerer) [inline lib/tasks/clean_up_submissions.rake]: You will remove `ApiKeys`. You can probably limit this to `Api.publishable_models`.

File: lib/tasks/clean_up_submissions.rake [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+      if m.column_names.include? 'user_id'
+        records = m.where("user_id IS NOT NULL")
+        puts "Found #{records.length} records of type #{m.name} with user_id not set to null. Purging."
+        records.each do |r|
+          r.delete
```
Comment (kammerer) [inline lib/tasks/clean_up_submissions.rake]: Why `r.delete` on each record and not `records.delete_all`? I assume that using `records.destroy_all` is too slow in some cases, so deletion is probably OK.

File: lib/tasks/clean_up_submissions.rake (hunk 1 above)
Comment (kammerer) [inline lib/tasks/clean_up_submissions.rake]: What's more, it does not seem to remove deleted records from the index.

**Comment by kammerer** [comment]
@nowakowski Please, make sure API keys are not deleted and that ES indices are properly updated (probably a rake task is the easiest option).

File: lib/tasks/clean_up_submissions.rake (hunk 1 above)
Comment (nowakowski) [inline lib/tasks/clean_up_submissions.rake]: I took this from https://www.elastic.co/guide/en/elasticsearch/reference/current/indices-refresh.html - could you suggest a better command?

File: lib/tasks/clean_up_submissions.rake (hunk 2 above)
Comment (nowakowski) [inline lib/tasks/clean_up_submissions.rake]: Okay.

**Comment by nowakowski** [comment]
@kammerer When speaking of rake tasks, do you mean `elasticsearch:import:all`?

**Comment by kammerer** [comment]
@nowakowski Yes, I meant this one (provided it rebuilds indexes from scratch). Another option (taken from specs/rails_helper.rb) is: 
```
models.each do |model|                                                                                                       
  model.import force: true, refresh: true                                                                                               
end
```

**Comment by nowakowski** [comment]
@kammerer Your requests have been followed up on.

**Comment by kammerer** [comment]
It did not work due to FK violation. Fixed.

**Comment by kammerer** [comment]
:+1:

**Comment by Nuanda** [comment]
@nowakowski Piotr, please merge, deploy, dump the prod db, run the script, repent your sins.

**Comment by nowakowski** [comment]
In progress.

**Comment by nowakowski** [comment]
Done (except the repentance bit ;).

**Pull request merged**